mailmerge/template_message.py

```diff
@@ -85,38 +85,67 @@ def _transform_recipients(self):
 
     def _make_message_multipart(self):
```

Comment:
I took this PR as an opportunity to refactor and comment this function.  I believe you originally wrote this function, would you mind double-checking my work?